Backported patch to allow us to use opencore-amr libraries.
==========================================================================
--- a/libavcodec/libamr.c
+++ b/libavcodec/libamr.c
@@ -531,7 +531,9 @@
 #define typedef_h
 #endif
 
+#if CONFIG_LIBAMR_WB_ENCODER
 #include <amrwb/enc_if.h>
+#endif
 #include <amrwb/dec_if.h>
 #include <amrwb/if_rom.h>
 
@@ -576,6 +578,7 @@
     Word16 allow_dtx;
 } AMRWBContext;
 
+#if CONFIG_LIBAMR_WB_ENCODER
 static int amr_wb_encode_init(AVCodecContext * avctx)
 {
     AMRWBContext *s = avctx->priv_data;
@@ -633,6 +636,7 @@
     size = E_IF_encode(s->state, s->mode, data, frame, s->allow_dtx);
     return size;
 }
+#endif
 
 static int amr_wb_decode_init(AVCodecContext * avctx)
 {
@@ -702,6 +706,7 @@
     .long_name = NULL_IF_CONFIG_SMALL("libamr-wb Adaptive Multi-Rate (AMR) Wide-Band"),
 };
 
+#if CONFIG_LIBAMR_WB_ENCODER
 AVCodec libamr_wb_encoder =
 {
     "libamr_wb",
@@ -715,5 +720,6 @@
     .sample_fmts = (enum SampleFormat[]){SAMPLE_FMT_S16,SAMPLE_FMT_NONE},
     .long_name = NULL_IF_CONFIG_SMALL("libamr-wb Adaptive Multi-Rate (AMR) Wide-Band"),
 };
+#endif
 
 #endif //CONFIG_LIBAMR_WB
